embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/spans/CurrentSessionSpanImpl.kt

```diff
@@ -98,7 +98,9 @@ internal class CurrentSessionSpanImpl(
                 if (appTerminationCause == null) {
                     endingSessionSpan.stop()
                     spanRepository.clearCompletedSpans()
-                    sessionSpan.set(startSessionSpan(openTelemetryClock.now().nanosToMillis()))
+                    if (startNewSession) {
```

Comment:
This introduces undesirable behavior where calling `sessionSpan.get()` afterwards means folks could attempt to write to the span after the session has ended. Setting the reference to `null` if no new session is created should be enough to fix it?

Reply:
Good call. Will fix tomorrow!

Reply:
This brings up an interesting point about what happens if we log traces when there is no session span - or do anything when there is no session span.

Before OTel, that data is just lost, right?

Reply:
I'll put up the fix for this but lets figure this out before I merge this - we should have tests to verify this case and ensure we don't crash

Reply:
Added tests to verify no-session-span state. This should be good to go.